src/alire/alire-pinning.ads

```diff
@@ -19,13 +19,22 @@ package Alire.Pinning is
    --  must exist in the solution. Root dependencies are given, and a previous
    --  solution with possibly more pins. The resulting solution may be invalid.
 
+   function Pin_To (URL      : String;
+                    Solution : Solutions.Solution;
+                    Crate    : Crate_Name)
+                    return Solutions.Solution with
+     Pre => Solution.Depends_On (Crate) or else
+     raise Checked_Error with
+       "Cannot pin crate not in dependencies: " & (+Crate);
```

Comment:
It is checked [here](https://github.com/mosteo/alire/blob/81deb84f2f0e9452225164903779901077977348/src/alr/alr-commands-pin.adb#L100). However your question puts the spot on something I've wondered a few times: should we treat `-gnata` as a requisite (hence you can rely on these checks -- incidentally we are doing this for `Alire.Crate_Name`) or not (so they're only useful for testing -- and you want the Pre/Post never to trigger).

I don't really like the second option unless one is using SPARK, or it's a explicit codebase policy. I have never seen this discussed from this perspective either, excepts some remarks from Kazakov at `c.l.a` that I couldn't really follow to conclusion. If checks are expected to be disabled at some point, certainly it makes no sense to use "expected" exceptions like `Checked_Error` in our case, since that changes the program from valid to erroneous.

Do you have some policy about this at AdaCore? (I.e., checks are always disabled for release builds).

Reply:
I don't think we have a common policy for releases. It's all a question of performances.

For the first releases I think we should enable all checks, so that we catch as many problems as possible. But then we might want to remove them for performance.

Reply:
Ok, so better not rely on Pre/Post checks as part of the "actual" code for the future.